### PR TITLE
[kube-state-metrics] New helm chart

### LINKED ISSUE #839: [kube-state-metrics] Migration to this repository
**Is your feature request related to a problem? Please describe.**
The maintainers (@tariq1890, @mrueg) of the kubernetes/kube-state-metrics chart would like to be host it in this repository.
Currently the chart is part of the repository which causes issues to non-helm users as the chart creates a lot of releases (cf. https://github.com/kubernetes/kube-state-metrics/issues/1392).

**Describe the solution you'd like**
Starting with kube-state-metrics release 2.0, we would like to create a PR to host the chart in this repository including its history and maintain it here (if additional maintainers are interested, hit Tariq or myself up on kubernetes' slack instance). The old chart is considered deprecated and eventually will be removed from the kube-state-metrics repository.

**Describe alternatives you've considered**
We have considered maintaing the chart in lockstep with the application, due to the nature of kube-state-metrics this is not an ideal solution.

### PR DESCRIPTION
**Pull request by mrueg**
#### What this PR does / why we need it:
This migrates the kube-state-metrics helm chart to this repository as mentioned in https://github.com/prometheus-community/helm-charts/issues/839 It includes the history of commits from helm/charts and kubernetes/kube-state-metrics.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #839

#### Special notes for your reviewer:
Please have a look @scottrigby @tariq1890 

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by monotek — COMMENTED** [review]
@prometheus-community/helm-charts 

Please review :)

File: charts/kube-state-metrics/LICENSE
Comment (desaintmartin) [inline charts/kube-state-metrics/LICENSE:1]: Probably remove it since it is the same as the one at root!

File: charts/kube-state-metrics/OWNERS
Comment (desaintmartin) [inline charts/kube-state-metrics/OWNERS:2]: @tariq1890 @mrueg are you still ok to maintain the chart?

File: charts/kube-state-metrics/README.md
Comment (desaintmartin) [inline charts/kube-state-metrics/README.md:20]: Maybe remove the helm2 instructions?

**Review by desaintmartin — COMMENTED** [review]
I suppose the other parts of the chart are copied from stable! Thanks!

File: charts/kube-state-metrics/OWNERS
Comment (mrueg) [reply charts/kube-state-metrics/OWNERS:2]: I'd like to keep maintaining it (I think @tariq1890 as well). We are open if others are interested as well.

File: charts/kube-state-metrics/LICENSE
Comment (mrueg) [reply charts/kube-state-metrics/LICENSE:1]: Will do!

File: charts/kube-state-metrics/README.md
Comment (mrueg) [reply charts/kube-state-metrics/README.md:20]: Once this has been in, I'll work on removing it (see: https://github.com/kubernetes/kube-state-metrics/pull/1358 )

**Comment by mrueg** [comment]
> I suppose the other parts of the chart are copied from stable! Thanks!

Yes, commits come from helm/stable and its current location in kubernetes/kube-state-metrics. Both had DCO (helm) / CLA (kubernetes) requirements. CI is failing due to that (as it only looks for signed-off-by I think), not sure how to resolve that.

File: charts/kube-state-metrics/OWNERS
Comment (tariq1890) [reply charts/kube-state-metrics/OWNERS:2]: Yes, I'd still be interesting in maintaining the helm chart

File: charts/kube-state-metrics/README.md
Comment (desaintmartin) [reply charts/kube-state-metrics/README.md:20]: What is the plan, removing the chart from kube-state-metrics.git once this is merged?

File: charts/kube-state-metrics/LICENSE
Comment (desaintmartin) [reply charts/kube-state-metrics/LICENSE:1]: In a later PR ?

**Comment by desaintmartin** [comment]
Regarding merging, we could do it manually (after at least one other review!). What was the procedure for the previous chart migrations?

**Comment by monotek** [comment]
> Regarding merging, we could do it manually (after at least one other review!). What was the procedure for the previous chart migrations?

I think @torstenwalter or @scottrigby had to disable the "only suquash merge" setting temporarly so the single commits could be preserved.

**Review by monotek — DISMISSED** [review]

**Review by gkarthiks — DISMISSED** [review]
/lgtm

**Comment by gkarthiks** [comment]
@scottrigby I guess the old commits are missing the sign-off. considering the commit history and nature of it (new chart donation into prom-community), should we have an exception for this?

PS: not mentioning the chart-admins. Please feel free to chime in.

**Comment by monotek** [comment]
I guess @brancz could help to approve / merge, while keeping git history, too.

File: charts/kube-state-metrics/LICENSE
Comment (mrueg) [reply charts/kube-state-metrics/LICENSE]: I removed it in this one. :) Thanks for the reminder!

**Comment by mrueg** [comment]
Added another commit to remove the license and rebased on main branch.

**Review by monotek — APPROVED** [review]

**Comment by scottrigby** [comment]
> @scottrigby I guess the old commits are missing the sign-off. considering the commit history and nature of it (new chart donation into prom-community), should we have an exception for this?
> 
> PS: not mentioning the chart-admins. Please feel free to chime in.

Thanks yes, I'm sorry it took me a while to reply. I have been heads down.

The DCO issue was discussed at length here, and finally approved https://github.com/kubernetes/kube-state-metrics/pull/1237. It's the same situation here, therefore I have manually set DCO bot to pass on this PR, and am approving it 👍

**Review by scottrigby — APPROVED** [review]

**Comment by scottrigby** [comment]
Thank you @mrueg 🏆

**Comment by scottrigby** [comment]
For this PR I am enabling "Allow merge commits" option, so that all the previous git history will be retained in `main`.

**Comment by scottrigby** [comment]
Disabled "Allow merge commits" option until needed again.

**Pull request merged**
